backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -464,10 +526,24 @@ private void getOrCreateInternalEndpointCommonChecks(RequestSystemSubscriptionPr
     @GET
     @Path("/{id}")
     @Produces(APPLICATION_JSON)
-    @RolesAllowed(ConsoleIdentityProvider.RBAC_READ_INTEGRATIONS_ENDPOINTS)
     @Operation(summary = "Retrieve an endpoint", description = "Retrieves the public information associated with an endpoint such as its description, name, and properties.")
     public EndpointDTO getEndpoint(@Context SecurityContext sec, @PathParam("id") UUID id) {
-        String orgId = getOrgId(sec);
+        if (this.backendConfig.isKesselBackendEnabled()) {
+            this.kesselAuthorization.hasPermissionOnResource(sec, IntegrationPermission.VIEW, ResourceType.INTEGRATION, id.toString());
+
+            return this.internalGetEndpoint(sec, id);
+        } else {
+            return this.legacyGetEndpoint(sec, id);
+        }
+    }
+
+    @RolesAllowed(ConsoleIdentityProvider.RBAC_READ_INTEGRATIONS_ENDPOINTS)
+    public EndpointDTO legacyGetEndpoint(final SecurityContext securityContext, final UUID id) {
+        return this.internalGetEndpoint(securityContext, id);
+    }
+
+    public EndpointDTO internalGetEndpoint(final SecurityContext securityContext, final UUID id) {
```

Comment:
why `public`? same comment for all *internal* methods

Reply:
Yes, no specific reason. I've changed them to `private`!

Reply:
Apparently the ones that have the `@Valid` or the `@RolesAllowed` annotations need to be at least `protected`... so I'm applying that scope to the methods that allow it and the `private` one to the ones that do not have any of those annotations.